packages/components/src/shared/src/useDisableInteraction.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * useDisableInteraction provides the original event handler but disables interaction
+ * if the boolean passed is true.
+ */
+import { useCallback, MouseEventHandler } from "react";
+
+export function useDisableClick<T = Element>(disabled?: boolean, handleClick?: MouseEventHandler<T>) {
```

Comment:
Since there is only a single function in the file, I would recommend matching the name of file and the name of the function.